TestStepDeleteExistingLabelMatch flake

Happens when using `go test -count 100`
```
2025-11-19T07:51:04.4722500Z === RUN   TestStepDeleteExistingLabelMatch
2025-11-19T07:51:04.4722603Z     step_integration_test.go:315:
2025-11-19T07:51:04.4722934Z            Error Trace:    /home/runner/work/kuttl/kuttl/internal/step/step_integration_test.go:315
2025-11-19T07:51:04.4723052Z            Error:          Received unexpected error:
2025-11-19T07:51:04.4723150Z                            namespaces "world" already exists
2025-11-19T07:51:04.4723263Z            Test:           TestStepDeleteExistingLabelMatch
2025-11-19T07:51:04.4723382Z --- FAIL: TestStepDeleteExistingLabelMatch (0.00s)
```

--- a/internal/step/step_integration_test.go
+++ b/internal/step/step_integration_test.go
@@ -233,7 +233,7 @@ func TestCheckResourceIntegration(t *testing.T) {
 
 // Verify that the DeleteExisting method properly cleans up resources that are matched on labels during a test step.
 func TestStepDeleteExistingLabelMatch(t *testing.T) {
-	namespace := "world"
+	namespace := fmt.Sprintf("kuttl-test-%s", petname.Generate(2, "-"))
 
 	podSpec := map[string]interface{}{
 		"containers": []interface{}{
@@ -244,15 +244,15 @@ func TestStepDeleteExistingLabelMatch(t *testing.T) {
 		},
 	}
 
-	podToDelete := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("aa-delete-me", "world"), map[string]string{
+	podToDelete := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("aa-delete-me", namespace), map[string]string{
 		"hello": "world",
 	}), podSpec)
 
-	podToKeep := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("bb-dont-delete-me", "world"), map[string]string{
+	podToKeep := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("bb-dont-delete-me", namespace), map[string]string{
 		"bye": "moon",
 	}), podSpec)
 
-	podToDelete2 := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("cc-delete-me", "world"), map[string]string{
+	podToDelete2 := kubernetes.WithSpec(t, kubernetes.WithLabels(t, kubernetes.NewPod("cc-delete-me", namespace), map[string]string{
 		"hello": "world",
 	}), podSpec)
 

